nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/DebugFlow.java

```diff
@@ -0,0 +1,414 @@
+                        FF_FAILURE_CURR = 0;
+                        curr_ff_resp = curr_ff_resp.getNextCycle();
+                    }
+                }
+                if (curr_ff_resp == FlowfileResponse.FF_ROLLBACK_RESPONSE) {
+                    if (FF_ROLLBACK_CURR < FF_ROLLBACK_MAX) {
+                        FF_ROLLBACK_CURR += 1;
+                        logger.info("DebugFlow rolling back (no penalty) file={} UUID={}",
+                                new Object[]{ff.getAttribute(CoreAttributes.FILENAME.key()),
+                                        ff.getAttribute(CoreAttributes.UUID.key())});
+                        session.rollback();
+                        session.commit();
+                        break;
+                    } else {
+                        FF_ROLLBACK_CURR = 0;
+                        curr_ff_resp = curr_ff_resp.getNextCycle();
+                    }
+                }
+                if (curr_ff_resp == FlowfileResponse.FF_YIELD_RESPONSE) {
+                    if (FF_YIELD_CURR < FF_YIELD_MAX) {
+                        FF_YIELD_CURR += 1;
+                        logger.info("DebugFlow yielding file={} UUID={}",
+                                new Object[]{ff.getAttribute(CoreAttributes.FILENAME.key()),
+                                        ff.getAttribute(CoreAttributes.UUID.key())});
+                        session.rollback();
+                        context.yield();
+                        return;
+                    } else {
+                        FF_YIELD_CURR = 0;
+                        curr_ff_resp = curr_ff_resp.getNextCycle();
+                    }
+                }
+                if (curr_ff_resp == FlowfileResponse.FF_PENALTY_RESPONSE) {
+                    if (FF_PENALTY_CURR < FF_PENALTY_MAX) {
+                        FF_PENALTY_CURR += 1;
+                        logger.info("DebugFlow rolling back (with penalty) file={} UUID={}",
+                                new Object[]{ff.getAttribute(CoreAttributes.FILENAME.key()),
+                                        ff.getAttribute(CoreAttributes.UUID.key())});
+                        session.rollback(true);
+                        session.commit();
+                        break;
+                    } else {
+                        FF_PENALTY_CURR = 0;
+                        curr_ff_resp = curr_ff_resp.getNextCycle();
+                    }
+                }
+                if (curr_ff_resp == FlowfileResponse.FF_EXCEPTION_RESPONSE) {
+                    if (FF_EXCEPTION_CURR < FF_EXCEPTION_MAX) {
+                        FF_EXCEPTION_CURR += 1;
+                        logger.info("DebugFlow throwing NPE file={} UUID={}",
+                                new Object[]{ff.getAttribute(CoreAttributes.FILENAME.key()),
+                                        ff.getAttribute(CoreAttributes.UUID.key())});
+                        throw new NullPointerException("forced by " + this.getClass().getName());
```

Comment:
I have broaden it out to any RuntimeException.  Other exceptions will be expected to be checked causing compilation errors.